x-pack/legacy/plugins/ml/server/models/data_frame/transforms.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { callWithRequestType } from '../../../common/types/kibana';
+import {
+  DATA_FRAME_TRANSFORM_STATE,
+  DataFrameTransformEndpointData,
+} from '../../../public/data_frame/pages/transform_management/components/transform_list/common';
+
+export function transformServiceProvider(callWithRequest: callWithRequestType) {
+  async function deleteTransform(transformId: string) {
+    return callWithRequest('ml.deleteDataFrameTransform', { transformId });
+  }
+
+  async function stopTransform(options: {
+    transformId: string;
+    force: boolean;
+    waitForCompletion: boolean;
+  }) {
+    return callWithRequest('ml.stopDataFrameTransform', options);
+  }
+
+  async function startTransform(options: { transformId: string; force: boolean }) {
+    return callWithRequest('ml.startDataFrameTransform', options);
+  }
+
+  async function deleteTransforms(transformsInfo: DataFrameTransformEndpointData[]) {
+    const results = [];
```

Comment:
could be a `DataFrameTransformEndpointResultData[]`

Reply:
Updated in 5f0e7c3bf1c54e79732bc46364bef460a58400ab